source/common/matcher/matcher.h

```diff
@@ -211,8 +213,6 @@ class MatchTreeFactory : public OnMatchFactory<DataType> {
             std::make_pair(children.first, *MatchTreeFactory::createOnMatch(children.second)));
       }
 
-      auto on_no_match = createOnMatch(matcher.on_no_match());
-
```

Comment:
Nit: the style guide encourages declaring local variables as close to their first use as possible.

Reply:
It's used in both branches of `switch`. We'd have to duplicate the line to make it any closer. Is that recommended?

Reply:
Never mind then. I missed it's used in greater scope now.